jaxrs-base-server/src/test/java/io/telicent/smart/cache/server/jaxrs/resources/UserInfoResource.java

```diff
@@ -13,86 +13,52 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package io.telicent.smart.cache.server.jaxrs.resources;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import io.jsonwebtoken.*;
-import io.telicent.smart.cache.server.jaxrs.applications.MockKeyServer;
+import io.telicent.servlet.auth.jwt.JwtServletConstants;
 import jakarta.ws.rs.GET;
-import jakarta.ws.rs.HeaderParam;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.core.Context;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.Response;
 
-import java.nio.charset.StandardCharsets;
-import java.security.Key;
-import java.security.PublicKey;
-import java.util.Base64;
 import java.util.HashMap;
 import java.util.Map;
 
 
 @Path("/userinfo")
 public class UserInfoResource {
-    private static MockKeyServer keyServer;
-    public UserInfoResource() {
-    }
-
-    public static void setKeyServer(MockKeyServer keyServer) {
-        UserInfoResource.keyServer = keyServer;
-    }
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public Response getUserInfo(@HeaderParam("Authorization") String authorization) {
-        // Check if the token is there, just for testing purposes
-        if (authorization == null || !authorization.startsWith("Bearer ")) {
-            return Response.status(Response.Status.UNAUTHORIZED).entity("Missing token  or invalid").build();
+    @SuppressWarnings("unchecked")
+    public Response getUserInfo(@Context ContainerRequestContext request) {
+        Jws<Claims> jws = (Jws<Claims>) request.getProperty(JwtServletConstants.REQUEST_ATTRIBUTE_VERIFIED_JWT);
+        if (jws == null) {
```

Comment:
Realised I could massively simplify this aspect of the `MockKeyServer` by relying on the fact that when JWT auth is enabled the raw JWT is already placed into request attributes so no need to parse it out of the headers manually ourselves.